### ISSUE TITLE
types: quantileSeq expands types unnecessarily

### ISSUE THREAD

**Issue by domdomegg** [josdejong/mathjs#3197]
**Describe the bug**

The types for quantileSeq have an overly wide return type.

**To Reproduce**

```ts
// Type 'number | BigNumber | Unit | MathArray' is not assignable to type 'number'.
//   Type 'BigNumber' is not assignable to type 'number'.ts(2322)
const value: number = math.quantileSeq([1, 2, 3], 0.90);
```

I think value here should always be a number. This is the behaviour of for example `math.mean` and `math.median`.

**Comment by josdejong** [comment]
Thanks for your input, we could indeed improve the TypeScript definitions I think. Help would be welcome.

**Comment by josdejong** [comment]
Fixed now in `v12.4.3`

**Comment by Loksly** [comment]
Hi. This fix https://github.com/josdejong/mathjs/pull/3198/files broke basic usage:

Now this basic usage is broken.
```ts
const quantiles: number[] = math.quantileSeq([1, 2, 3, 4, 5], [0.25, 0.75]) as number[];
```
This was supported previous to v12.4.3.

**Comment by domdomegg** [comment]
Thanks @Loksly. I've raised a PR to fix this: https://github.com/josdejong/mathjs/pull/3223

In the mean time this workaround should work:

```ts
const quantiles: number[] = math.quantileSeq([1, 2, 3, 4, 5], [0.25, 0.75]) as unknown as number[];
```

**Comment by josdejong** [comment]
This fix is now published in `v13.0.2`
